activerecord/test/cases/locking_test.rb

```diff
@@ -738,6 +738,35 @@ def test_with_lock_rolls_back_transaction
       assert_equal old, person.reload.first_name
     end
 
+    def test_with_lock_configures_transaction
+      person = Person.find 1
+      Person.transaction do
+        outer_transaction = Person.connection.transaction_manager.current_transaction
+        assert_equal true, outer_transaction.joinable?
+        person.with_lock(requires_new: true, joinable: false) do
+          current_transaction = Person.connection.transaction_manager.current_transaction
+          assert_not_equal outer_transaction, current_transaction
+          assert_equal false, current_transaction.joinable?
+        end
+      end
+    end
+
+    def test_with_lock_forwards_lock_type
+      person = Person.find 1
+      assert_called_with(person, :lock!, ["FOR UPDATE NOWAIT"]) do
```

Comment:
Looking into switching this to assert_sql, but it's db-adapter specific because sqlite3 elides these statements.